lib/enum/enum.rb

```diff
@@ -29,6 +28,20 @@ def delete_all
         super
       end
 
+      def define_enum_methods
+        return if @enum_accessors.blank?
+
+        @records&.each do |record|
+          enum_value = @enum_accessors.map { |name| record.attributes[name] }.join("_").downcase.gsub(/\W+/, '_').gsub(/^_|_$/, '')
+          method_name = "#{enum_value}?"
+          define_method(method_name) do
+            self.id == record.id
```

Comment:
Yea. Instead of Id, just compare the value. This will allow duplicates and work closer to rails.